### PR TITLE
chore: Add read-only user role

### PR DESCRIPTION
**Pull request by Thource**
### What changed

Added a read-only user role.

### Why did it change

Requirement has been confirmed for a PECS Read Only role for the purposes of HMPPS policy audit/assurance team(s). This has been approved by the Head of PECS. 

### Issue tracking

- P4-2842

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd></kbd> | <kbd></kbd> |

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### CODE REVIEW

File: common/lib/permissions.js [hunk 1]
```diff
@@ -43,6 +43,21 @@ const contractDeliveryManagerPermissions = [
   'youth_risk_assessment:print',
 ]
 
+const readOnlyPermissions = [
```
Comment (teneightfive) [inline common/lib/permissions.js:46]: Could we add a test to the [user library test](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/baec5e26e3dfb411ef2abaa7880f745cf7e437e6/common/lib/user.test.js#L121) (which is currently where these are unit tested).

File: common/lib/permissions.js [hunk 2]
```diff
@@ -43,6 +43,21 @@ const contractDeliveryManagerPermissions = [
...
+  'dashboard:view',
+  'allocations:view',
+  'locations:contract_delivery_manager',
```
Comment (teneightfive) [inline common/lib/permissions.js:49]: I can't remember exactly what this allows but I'm assuming the users with this role need to be able to see locations in this way? Does it allow them to see *all* locations?

File: common/lib/permissions.js (hunk 1 above)
Comment (Thource) [reply common/lib/permissions.js:46]: Oops, I searched for "contract_delivery_manager" and concluded that there were no tests :D

File: common/lib/permissions.js (hunk 2 above)
Comment (Thource) [reply common/lib/permissions.js:49]: `locations:contract_delivery_manager` is in a conditional that leads to getting all locations for all suppliers, `locations:all` gives permission to access the `/all` route

Comment (teneightfive) [reply common/lib/permissions.js:49]: And I'm guessing this is something required for this user type?

**Review by teneightfive — APPROVED** [review]

File: common/lib/permissions.js (hunk 2 above)
Comment (Thource) [reply common/lib/permissions.js:49]: I believe so, from the ticket: This should be a replica of the current PECS CDM role, with access to Incoming and Outgoing moves for all PECS locations, but without the ability to see the maps.

**Pull request merged**